lib/PreferenceList.cc

```diff
@@ -70,15 +70,15 @@ std::ostream& operator<<(std::ostream& out, const PreferenceList* pl) {
     std::stringstream stmp;
 
     for (auto i = pl->cbegin(), e = pl->cend(); i != e; ++i) {
-        if(i->vertex != nullptr){
+        if (i->vertex != nullptr) {
             stmp << i->vertex->get_id();
-        }
-        else{
-            int index = i-pl->cbegin();
+        } else {
+            auto index = i-pl->cbegin();
```

Comment:
index should be changed to use std::distance, i and pl->cbegin() are both iterators.